R/animint.R

```diff
@@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
 saveChunks <- function(x, vars, meta){
   if(is.data.frame(x)){
     if(length(vars) == 0){
+      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
+        if("group" %in% names(x)) {
+          # common info for groups, save only once
+          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
+          chunk.shape <- file.path(meta$out.dir, csv.name)
+          if(!file.exists(chunk.shape)){
+            x1 <- x[c("x", "y", "group")]
+            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE, 
+                        sep = "\t")
+          }
+          # info for each group
+          x2 <- x[, !names(x) %in% c("x", "y")]
+          x <- x2[!duplicated(x2), ]
+          }
```

Comment:
Which version of the WorldBank viz did you look at?

For the first example wb.all, there is a row for every (year,country)
combination
https://github.com/tdhock/animint/blob/master/inst/examples/WorldBank.R#L4
so each chunk has the same number of rows, and I guess it should be easy to
detect that color/clickSelects are the same across chunks.

For the other examples which use the not.na data set there are some rows
that have been removed, so each chunk does not have the same number of
rows, and I guess it would not be feasible to detect.

On Fri, Jun 12, 2015 at 6:07 AM, Tony Tsai <notifications@github.com> wrote:

> In R/animint.R
> <https://github.com/tdhock/animint/pull/76#discussion_r32304373>:
>
> > @@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
> >  saveChunks <- function(x, vars, meta){
> >    if(is.data.frame(x)){
> >      if(length(vars) == 0){
> > +      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
> > +        if("group" %in% names(x)) {
> > +          # common info for groups, save only once
> > +          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
> > +          chunk.shape <- file.path(meta$out.dir, csv.name)
> > +          if(!file.exists(chunk.shape)){
> > +            x1 <- x[c("x", "y", "group")]
> > +            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE,
> > +                        sep = "\t")
> > +          }
> > +          # info for each group
> > +          x2 <- x[, !names(x) %in% c("x", "y")]
> > +          x <- x2[!duplicated(x2), ]
> > +          }
>
> Actually in the WorldBank data viz, the color and clickSelects columns
> are different for chunks. For example, the colour are different at 78 79
> 81 84 85 86 87 88 89 90 91 93 94 95 96 97 98 99 101 rows for the first
> two chunks. @tdhock <https://github.com/tdhock> could you come up with
> another example for the case you mentioned?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tdhock/animint/pull/76/files#r32304373>.
>